### ISSUE TITLE
Extend example script to the 2D case.

### ISSUE THREAD

**Issue by Jez-Carter** [Jez-Carter/Bias_Correction#8]
Extend the following script (https://github.com/Jez-Carter/Bias_Correction/blob/main/scripts/Exploratory_Notebooks/1D%20Gaussian%20Process%20Examples.py) to the 2D case, this involves:

- [x] Creating a column of coordinates in the shape (n,2) - for the climate model these will come from a regular grid, while for the observations they'll come from an irregular grid and will be much more sparse (could again create a masked region to really highlight difference?). 
- [x] Adjusting the initial data plot to 3 subplots showing scatter plot representations of the data (could plot underlying output (from high-resolution sampled GP) as contour and points for fitting as circles? - like in:https://github.com/Jez-Carter/Bias_Correction/blob/main/results/Examining_Results/Examining%20GP%20Output%20Lima.ipynb)
- [x] I don't think the actual model function or the posterior predictive functions will need updating (just the inputs) but verify this.
- [x] Adjusting the posterior predictive plot to include 2 subplots of surfaces for the mean and variance, maybe with markers showing the locations of observations and climate data.

**Comment by Jez-Carter** [comment]
This issue is mostly complete and an example case of the 2D notebook has been committed to the repo (6423b06). One aspect still to examine is that the bias lengthscale is being estimated at 20 rather than the correct value of 40 - I believe this is due to the sparseness of observations and is not an issue with the code but it would be nice to double check this. It is noted that when extending to 2D the computational demands are in general greater and that it would be advantageous to look into ways of improving the speed of our code for fitting the model - I have created an issue for this (#17). 
- [ ] Explore model estimate of bias lengthscale.

**Comment by Jez-Carter** [comment]
I've explored the issue around under-predicting the lengthscale for the bias. It seems the issue was with the prior on the lengthscale, which was a Gamma(3,0.5) and so has very low probability around a value of 40. This combined with having a relatively small domain and so minimal features to determine the lengthscale from results in the lengthscale being very dependent on the prior and so tending towards lower values where Gamma(3,0.5) has greater probability. To fix this I've adjusted the prior for the bias to Gamma(0.001,0.001) (0ff4ba6), which should be very non-informative. The result is as expected with a much greater uncertainty in the lengthscale but a credible interval that includes the true value.
